### DISCUSSION TITLE
How to restart/reset refetchInterval?

### DISCUSSION THREAD

**Discussion by Jurek-Raben** [TanStack/query#5613] [General]
In our current project, we need to restart the refetchInterval of a query under specific conditions. The api does not seem to provide a way to do this currently. Or is there a way already? If not, please provide a way here, for example using the queryClient instance, like "queryClient.resetRefetchInterval(key)".

**Comment by TkDodo** [comment]
`refetchInterval` takes a function that gets the latest data passed, where you can decide, depending on that, if you want an interval or not.

**Comment by Jurek-Raben** [comment]
Sure, but I would like to restart the interval not dependent on this fetch, but another fetch(a post/mutation actually). So there is a fetch/query with a refetchInterval using various conditions in the refetchInterval method. And there is a post-button, if that button was clicked, I need the refetchInterval to be restarted. The refetchInterval method only makes a decision possible how long until refetch (based on the fetched data), but not a possible restart of the refetch counter.

**Reply by TkDodo** [reply]
if you invalidate the Query after the mutation, it will yield new data, in which case the `refetchInterval` function will run again.

**Reply by Jurek-Raben** [reply]
Will this also actually re-fetch? Because this is what I can't do. The post request needs some time on the server to be processed, and the fetch should be done later in some minutes.

So can I invalidate the fetch query, but without actually fetching, but instead using cache data?

**Reply by TkDodo** [reply]
Yes, it would refetch. You can't re-start an interval without actually fetching first.

**Reply by Jurek-Raben** [reply]
Ok, but that is what I need.

We have a post-endpoint which will update the fetch within some minutes (due using a complex multi server system which merges various data). So if the button was clicked, I need to make sure that the timeout is atleast bigger than X minutes (maximum time until the data is actually in the fetch). I think this kind scenario might be not very uncommon.

Could you consider to add this as a feature? Like queryClient.resetRefetchInterval(cacheId); or maybe better:
queryClient.getRefetchIntervalTime(cacheId)
queryClient.setRefetchIntervalTime(cacheId)

**Reply by TkDodo** [reply]
> Ok, but that is what I need.

Okay?

> We have a post-endpoint which will update the fetch within some minutes (due using a complex multi server system which merges various data). So if the button was clicked, I need to make sure that the timeout is atleast bigger than X minutes (maximum time until the data is actually in the fetch). I think this kind scenario might be not very uncommon.

What would be common is that you can just invalidate the query, which triggers a refetch. If your super slow complex server system hasn't processed data yet, it would respond with `{ done: false }` or `{ status: "processing" }` or whatever, in which case you can set up your interval fetching to try again some time later until the server is done.

> I need to make sure that the timeout is atleast bigger than X minutes

What do you do if your users reload the browser? Or open up the page in a new tab. Putting up the constraint that "no fetch can happen in the next X minutes" is kinda impossible ...

> Could you consider to add this as a feature?

no. interval refetching is an observer level feature, not a cache level feature. It doesn't work that way.

**Reply by Jurek-Raben** [reply]
That is not a bad idea. I'll try to convince the backend team, since we already have a pending state.
